packages/blockchain-wallet-v4-frontend/src/components/BuySell/Coinify/ISignThis/index.js

```diff
@@ -0,0 +1,168 @@
+import { actions } from 'data'
+import { prop } from 'ramda'
+import { Button, Text, TooltipHost, Icon } from 'blockchain-info-components'
+import { FormattedMessage } from 'react-intl'
+
+import renderFaq from 'components/FaqDropdown'
+import CountdownTimer from 'components/Form/CountdownTimer'
+import * as Currency from 'blockchain-wallet-v4/src/exchange/currency'
+import media from 'services/ResponsiveService'
+import { getData } from './selectors'
+
+const ISXContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  ${media.mobile`
+    flex-direction: column;
+  `};
+`
+const ButtonContainer = styled.div`
+  margin-left: 5%;
+  width: 20%;
+  ${media.mobile`
+    width: 100%;
+    margin-left: 0%;
+    margin-top: 20px;
+  `};
+`
+const TimerContainer = styled.div`
+  width: 66%;
+  padding-bottom: 5px;
+  display: flex;
+  justify-content: flex-end;
+`
+const QuoteExpiredText = styled(Text)`
+  span {
+    margin-right: 5px;
+  }
+  span:first-of-type {
+    font-style: italic;
+  }
+`
+const IframeWrapper = styled.div`
+  width: 65%;
+  ${media.mobile`
+    width: 100%;
+  `};
+`
+const ISignThisIframe = styled.iframe`
+  width: 100%;
+  height: 400px;
+  border: ${props => `1px solid ${props.theme['gray-1']}`};
+`
+
+const tradeQuestions = [
+  {
+    question: (
+      <FormattedMessage
+        id='scenes.coinify.isx.tradequestion'
+        defaultMessage='How is my payment method used?'
+      />
+    ),
+    answer: (
+      <FormattedMessage
+        id='scenes.coinify.isx.tradeanswer'
+        defaultMessage="Bitcoin is delivered to your Blockchain wallet by using the information you've provided. With that in mind, please double check that your submitted details are error-free. If you pay by credit / debit card, your bitcoin will be delivered within a couple of hours after the transaction is completed, depending on your bank’s transfer policies. If you pay by bank transfer, your bitcoin will be delivered after Coinify has processed your order, which usually takes between 2-3 days."
+      />
+    )
+  }
+]
+const getExpiredBtcValues = q =>
+  q.quoteCurrency === 'BTC' ? `${q.quoteAmount / 1e8}` : `${q.baseAmount / 1e8}`
+const getExpiredFiatValues = q =>
+  q.baseCurrency !== 'BTC'
+    ? `${Currency.formatFiat(Math.abs(q.baseAmount))} ${q.baseCurrency}`
+    : `${Currency.formatFiat(Math.abs(q.quoteAmount))} ${q.quoteCurrency}`
+
+class ISignThisContainer extends PureComponent {
+  state = {
+    quoteExpired: false
+  }
+
+  onQuoteExpiration = () => {
+    this.setState({ quoteExpired: true })
+  }
+
+  render () {
+    const { iSignThisId, coinifyActions, trade, quoteR } = this.props
+    // NOTE: iSignThisId is now a payment URL for the new payment provider
+    // see here: https://github.com/blockchain/bitcoin-coinify-client/pull/32/files#diff-a17a0bd50a25f6c13d0f93c7155ca029R144
+    const srcUrl = iSignThisId
+
+    return (
+      <Fragment>
+        <TimerContainer>
+          {quoteR
+            .map(q => {
```

Comment:
no keys for top-level components
single char variable makes look up until line above this when looking for what `q` means, better change to `quote` imo